kik_unofficial/client.py

```diff
@@ -856,9 +856,21 @@ def data_received(self, data: bytes):
             self.loop.call_soon_threadsafe(self.api._on_new_data_received, data)
             return
 
-        # Handle 'ack' packets with 'pong'
-        if re.search(b'<ack id="[^"]*"/><pong/>', data):
-            self.loop.call_soon_threadsafe(self.api._on_new_data_received, data)
+        # remove ack packets from data Sometimes they can surround other data and make it useless.
```

Comment:
A better solution to this problem would be to change `on_new_data_received` code to iterate the received XML elements and ignore the `ack`s. If you choose to use the regex aproach, I think we can make a shorter comment than this one to show the problem. Something like `<ack id=.... /><iq id=...></iq>`